app/models/concerns/team_duplication.rb

```diff
@@ -7,215 +7,109 @@ module TeamDuplication
     end
 
-    def self.set_mapping(object, copy)
-      key = object.class_name.to_sym
-      @mapping[key] ||= {}
-      @mapping[key][object.id] = copy
+    def self.alter_project_copy(copy)
+      copy.generate_token(true)
+      copy.set_slack_notifications_enabled = false
     end
 
-    def self.copy_image(original, copy)
-      [:logo, :lead_image, :file].each do |image|
-        next unless original.respond_to?(image) && original.respond_to?("#{image}=") && original.send(image)
-        [original.send(image)].flatten.each do |img|
-          copy.send("#{image}=", open(img.url)) if img.url =~ /^#{CONFIG['storage']['endpoint']}/
-        end
+    def self.alter_copy_by_type(original, copy)
+      if original.is_a?(Team)
+        self.alter_team_copy(copy)
+      elsif original.is_a?(Project)
+        self.alter_project_copy(copy)
+      elsif original.is_a?(TagText)
+        copy.team_id = @team_id if !@team_id.nil?
       end
-    end
-
-    def self.versions_log_mapping(original, copy)
-      if original.respond_to?(:get_versions_log)
-        original.get_versions_log.find_each do |log|
-          self.set_mapping(log, copy)
-        end
+      copy.save(validate: false)
+      if original.is_a?(Project)
+        @project_id_map[original.id] = copy.id
+      elsif copy.is_a?(Team)
+        @team_id = copy.id
       end
     end
 
-    def self.copy_annotations
-      [:ProjectMedia, :Source, :Task].each do |type|
-        next if @mapping[type].blank?
-        @mapping[type].each_pair do |original, copy|
-          type.to_s.constantize.find(original).annotations.find_each do |a|
-            a = a.load
-            annotation = a.dup
-            annotation.annotated = copy
-            annotation.is_being_copied = true
-            annotation.skip_notifications = true
-            Team.copy_image(a, annotation)
-            annotation.save(validate: false)
-            self.set_mapping(a, annotation)
-            self.copy_annotation_fields(a, annotation)
+    def self.update_team_rules(new_team)
+      (new_team.get_rules||[]).each do |rule|
+        (rule["actions"]||[]).each do |action|
+          if action["action_definition"] == "move_to_project" || action["action_definition"] == "add_to_project"
+            action["action_value"] = @project_id_map[action["action_value"].to_i].to_s
           end
         end
       end
+      new_team
     end
 
-    def self.copy_annotation_fields(original, copy)
-      original.get_fields.each do |f|
-        next unless f.is_a?(DynamicAnnotation::Field)
-        field = f.dup
-        field.annotation_id = copy.id
-        field.save(validate: false)
-        self.set_mapping(f, field)
+    def self.process_team_users(t, team, processed_user_ids)
+      t.team_users.each do |tu|
+        next if processed_user_ids.include?(tu.user_id)
+        new_tu = TeamUser.new(tu.attributes.select{|k,_| k!="id"})
+        new_tu.team = team
+        new_tu.save!
       end
     end
```

Comment:
If the idea of this method is to migrate the `TeamUser`s, we don't need to - only the `TeamBotInstallation`s should be duplicated. And the user that is performing the duplication should be the owner of the new team.